### PR TITLE
fix(VolumeMapper): Support fractional or proportional opacity mode

### PR DESCRIPTION
**Pull request by scottwittenburg**
Adds support for an `opacityMode` on the `VolumeProperty`, which is only used in the shader if `independentComponents` is turned on.  If `opacityMode` is set to `proportional`, then we can also set a `proportionalComponent`, and then the opacity contribution from that component will be ignored in the shader.

### CODE REVIEW

**Comment by scottwittenburg** [comment]
@thewtex Here's a first pass at the changes we talked about for the `VolumeMapper`.

**Comment by scottwittenburg** [comment]
@thewtex I added support for multiple proportional components, some description of the functionality, and a test.  I think this is now ready for review.

File: Sources/Rendering/OpenGL/VolumeMapper/test/testProportionalComponents.js [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+import vtkRenderWindow from 'vtk.js/Sources/Rendering/Core/RenderWindow';
+import vtkRenderWindowInteractor from 'vtk.js/Sources/Rendering/Core/RenderWindowInteractor';
+import vtkRenderer from 'vtk.js/Sources/Rendering/Core/Renderer';
+import vtkVolume from 'vtk.js/Sources/Rendering/Core/Volume';
+import vtkVolumeMapper from 'vtk.js/Sources/Rendering/Core/VolumeMapper';
+import { OpacityMode } from 'vtk.js/Sources/Rendering/Core/VolumeProperty/Constants';
+
+import baseline1 from './testProportionalComponents.png';
+
+test.onlyIfWebGL('Test Volume Rendering with Proportional Component', (t) => {
```
Comment (thewtex) [inline Sources/Rendering/OpenGL/VolumeMapper/test/testProportionalComponents.js:18]: cool

File: Sources/Rendering/OpenGL/VolumeMapper/test/testProportionalComponents.js [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+  // Interactor
+  const interactor = vtkRenderWindowInteractor.newInstance();
+  interactor.setStillUpdateRate(0.01);
+  interactor.setView(glwindow);
+  interactor.initialize();
+  interactor.bindEvents(renderWindowContainer);
+
+  renderer.addVolume(actor);
+  renderer.resetCamera();
+  console.log('camera setup');
```
[the remark this answers is not in the document]
Comment (scottwittenburg) [reply Sources/Rendering/OpenGL/VolumeMapper/test/testProportionalComponents.js]: I don't know what it is with me and debug logging.  Sigh...

**Pull request merged**